release_notes.txt

```diff
@@ -1,3 +1,73 @@
+
+May 14, 2018
+
+New Features:
+--------------
+- Drivers have new `debug_print` option that writes design vars, objective, and constraints to stdout for each iteration.
+- Added DOEDriver for serial and parallel cases to the standard library.
+- Solvers have new `debug_print` option that will report the initial condition of the solver if it fails to converge.
+- SimpleGAdriver will now compute a population size based on 4*sum(bits) + 1 if pop_size=0
+- New problem methods `<problem>.get_val()` and `<problem>.set_val()` allow the user to specify a unit, and we'll convert to the correct unit (or throw an error if the unit is not compatible).
+- OpenMDAO command line utility now works under MPI.
+- `list_problem_vars()` method on problem will report all design vars, objectives, and constraints.
+- ExternalCodeComp has new option, `allowed_return_codes` to specify valid return codes.
+
+- New Analytic Derivatives Features
+    - Major re-write of the Feature documentation on the Core Feature, "Working with Analytic Derivatives."
+    - User can now assign AssembledJacobians. (Big speedup for a lot of lower-order or serial problems).
+    - Automatic calculation of total Jacobian sparsity for optimizers that support it (lower compute cost when using SNOPT).
+    - Automatic simultaneous coloring algorithm for separable problems to speed up serial derivatives calculations.
+    - Option for caching of linear solution vectors to speed up linear solves when using iterative solvers
+
+- New Components in the standard library:
+    - CrossProductComp
+    - DotProductComp
+    - MatrixVectorProductComp
+
+- Updates to existing Components in standard library:
+    - LinearSystemComp has been vectorized.
+    - KSComp has been vectorized.
+    - BsplineComp has been vectorized.
+
+- New CaseRecording/CaseReading Features:
+    - CaseRecorder now saves metadata/options from all groups and components.
+    - CaseReader now supports `list_inputs` and `list_outputs` methods.
+    - User can iterate over cases from a CaseReader in either flat or hierarchical modes.
+    - User can re-load a case back into a model via the `load_case()` method.
+
+
+Backwards-Compatible API Changes:
+-----------------------------------
+- Changed all instances of `metadata` to `options`. `<system>.metadata` has now been deprecated
+
+- Class name changes to Components, so all end in "Comp" (old class names have been deprecated):
+    - MetaModelStructured -> MetaModelStructuredComp
+    - MetaModelUnStructured -> MetaModelUnStructuredComp
+    - MultiFiMetaModelUnStructured -> MultiFiMetaModelUnStructuredComp
+    - ExternalCode -> ExternalCodeComp
+
+- Driver metadata/options can now be set via init args (can still be set as options as well).
+- User no longer needs to specify PETSCvector for parallel cases (the framework figures it out now).
+
+- check_totals and check_partials no longer support the `supress_output` argument. Use `out_stream=None` instead.
+
```

Comment:
Also in `check_partials`, "comps" was removed and replaced by "inlucdes" and "excludes", which allow glob patterns.